Tests/ApolloTests/GraphQLMapEncodingTests.swift

```diff
@@ -3,9 +3,13 @@ import XCTest
 import ApolloTestSupport
 import StarWarsAPI
 
-class InputValueEncodingTests: XCTestCase {
+class GraphQLMapEncodingTests: XCTestCase {
+  private struct MockGraphQLMapConvertible: GraphQLMapConvertible {
+    let graphQLMap: GraphQLMap
+  }
   private func serializeAndDeserialize(_ map: GraphQLMap) -> NSDictionary {
-    let data = try! JSONSerializationFormat.serialize(value: map.withNilValuesRemoved)
+    let input = MockGraphQLMapConvertible(graphQLMap: map)
```

Comment:
I think you told me this at some point, but remind me why this wrapper is now necessary.

Reply:
I did not! haha.

It's not, but this test was testing that serialization happened correctly when the `GraphQLMap.withNilValuesRemoved` was called. But that is actually an implementation detail. The only place this is being used when we call `jsonValue` on a `GraphQLMapConvertible`. So two things

1) `withNilValuesRemoved` should really be `fileprivate` (which is now is in GraphQLMap.swift). The only reason it wasn't private before was so it could be called in this test.

2) And reason 1 should be an indicator that your test is not testing the actual intention. These tests should be testing that `GraphQLMapConvertible.jsonValue` returns the correct value to be serialized. Which includes removing the nil values. Currently, if we removed the `.withNilValuesRemoved` from the `jsonValue` computed property, these tests would still pass, but the production code would be broken. Because this test explicitly calls the same behavior we actually want, rather than verifying the behavior occurs. This PR fixes that. :)

Reply:
Yeah this is always a fun border issue when writing tests. Good catch.